### ISSUE TITLE
False positive "Unresolved reference".

### ISSUE THREAD

**Issue by alexander-irbis** [intellij-rust/intellij-rust#3058]
## Environment

* **Intellij-Rust plugin version:** Version: 0.2.0.2110-182
* **Rust toolchain version:** rustc 1.30.1 (1433507eb 2018-11-07)

## Problem description

Some methods of the `String` type are marked as "unresolved", while IDE may work with their results.
I guess this is related with generic arguments.

## Steps to reproduce

```rust
fn test() {
    let mut string = "23".to_string();
    if string.contains(".") {
        unreachable!();
    }
    string.insert(0, '1');
    let n: f64 = string.parse().unwrap();
}
```
![image](https://user-images.githubusercontent.com/1448323/48680402-d1642d00-eba3-11e8-8717-1e26cf2c25d3.png)

**Comment by vlad20012** [comment]
Hmm, can't reproduce it on the latest stable. Maybe some additional setup is required? Dependencies?
![image](https://user-images.githubusercontent.com/3221931/48681645-c960b900-ebb4-11e8-9a75-ad5415d35224.png)

**Comment by alxdb** [comment]
I'm also getting an unresolved error problem, I thought I'd put it here rather than starting a new issue as it seems related.

### Environment

- **IntelliJ Rust Version:** `0.2.0.2110-182`
- **Rust Version:** `1.30.1`

### Description

IntelliJ Rust incorrectly marks a method as unresolved, and it compiles without error with cargo.

I think this is related to having multiple traits that have methods with identical names. IntelliJ Rust identifies the incorrect trait, and if I import this, the error goes away. However Cargo now reports an error, as this is actually incorrect.

Specifically this occurs with itertools and specs which both define a trait method named `join`. IntelliJ Rust incorrectly identifies itertools as the trait implementor whereas it is supposed to be specs.

A minimum working example of specs can be found at [https://slide-rs.github.io/specs/02_hello_world.html#full-example-code](url). Copying this code into a new project and just adding the itertools crate as a dependency in Cargo.toml causes the problem in my case.

**Comment by alexander-irbis** [comment]
@alxdb Hi!
I believe this is a similar, but diffierent issue.
I can reproduce your issue on the other machine, where I can't reproduce my initial issue.
So I opened a new bug report #3083

**Comment by alexander-irbis** [comment]
Some notes on this issue and tries to fix.

## "Invalidate and Restart"

Didn't help.

## Change toolchain and then "Invalidate and Restart"

Didn't help.

## Upgrade Idea to 2018.3 and Rust plugin to 0.2.0.2111-183

Partially fixed: "unresolved reference" disappeared, the autocomplition became able to complete these methods, but methods highlighted as just unknown.

## Manually drop a content of ~/.IdeaIC2018/system/caches

The same as "Update"

## Create new clean profile

Fixed the issue.
But...

**Comment by AThilenius** [comment]
@alxdb Did you ever find a solution? I've tried invalidating, deleting `~/.IntelliJIdea2019.2\system\caches`, messing with imports. Exact same case as yours, trying to use `specs` and `itertools` together, in 2019.2.1 Ultimate with Rust plugin `0.2.104.2132-192`.

**Comment by alxdb** [comment]
Nope, unfortunately I never managed to find a solution. I ended up abandoning intelliJ for rust.

**Comment by AThilenius** [comment]
That's really disappointing 😔 I would love to have a good debugger and all the other cool things in Intellij.

**Comment by vlad20012** [comment]
@AThilenius @alxdb `specs`, `itertools` or any other crate issues are not related to this issue. Please, open a new issue with a detailed description and screenshot. I can't reproduce it with https://slide-rs.github.io/specs/02_hello_world.html#full-example-code (i.e. the listed code has errors, but it does not compile as is; IDEA suggest appropriate fixes, the code is successfully compiled when I applied them)

**Comment by AThilenius** [comment]
@vlad20012 Acknowledged. This is related to the issue @alxdb described, which is closest to #3083 that you opened, but that is also 'closed as fixed'. It has nothing to do with crate issues though, intellij-rust is failing to resolve Trait methods that `cargo build` has no problem with.

I apologize but I presently lack the time to investigate this more, which is why I haven't opened another issue with repro steps. Also apologies for necroing an old issue.

**Comment by yiv** [comment]
In my case, this problem was never fixed.
![err](https://user-images.githubusercontent.com/16640312/207524299-e91e325d-a992-4bee-b826-45a647647f5f.png)
